### PR TITLE
Fix troops sprites overlap in quick info

### LINKED ISSUE #7927: Adventure map, castle and hero right click info windows, creature sprites overlap
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

In fheroes2, creature sprites overlap in castle and hero right click info windows.

In fheroes2:

Creatures overlap 001

Creatures overlap 002

### Save file

Here is the save file:
[Info right clic buzz 01.zip](https://github.com/ihhub/fheroes2/files/12870605/Info.right.clic.buzz.01.zip)

### Additional info

Version 1.0.9 7923

### PR DESCRIPTION
**Pull request by Districh-ru**
Fix #7927

### CODE REVIEW

File: src/fheroes2/army/army_ui_helper.cpp [hunk 1]
```diff
@@ -44,49 +44,32 @@ void fheroes2::drawMiniMonsters( const Troops & troops, int32_t cx, const int32_
...
     }
 
+    const size_t slots = troops.Size();
     for ( size_t slot = 0; slot < slots; ++slot ) {
-        // Skip the monster handling when a slot is empty
-        if ( slotsToSkip > 0 && !isCompact ) {
-            --slotsToSkip;
-            cx -= chunk;
-            continue;
+        if ( count == 0 ) {
```
[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/army/army_ui_helper.cpp:62]: This is actually the correct statement so the other one is wrong.

**Review by ihhub — APPROVED** [review]

File: src/fheroes2/army/army_ui_helper.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Districh-ru) [reply src/fheroes2/army/army_ui_helper.cpp:63]: I changed `0 == count` (the old code) to `count == 0` and added comments for each condition not to confuse them.

**Comment by Branikolog** [comment]
Hi, @Districh-ru !
image

Are summoned elementals supposed to be placed to the left of the initial army? If I'm not mistaken in master branch summoned creatures appear to the right...
image

**Comment by Districh-ru** [comment]
> Hi, @Districh-ru !
> Are summoned elementals supposed to be placed to the left of the initial army? If I'm not mistaken in master branch summoned creatures appear to the right...

Hi, @Branikolog !
When there are lots of killed troops they are displayed in reverse order (from right to left) to better display overlaped troops' sprites (to show mostly heads instead of tails).
I can revert this reversing if showing summoned troops in the right part of dialog is imortant. :)

**Comment by Districh-ru** [comment]
Hi, @Branikolog, fixed:
изображение

Could you please check it when you have time.

**Review by Branikolog — APPROVED** [review]
@Districh-ru 
Great!

**Comment by ihhub** [comment]
@Districh-ru , many thanks for this fix!

**Comment by fheroes2bugs** [comment]
Is it fixed? 
image

I downloaded release and have this bug.
Or I should compile the game from the source.

I suggest just to update the release to 1.0.9.1 or so

**Comment by fheroes2bugs** [comment]
> > Is it fixed? image
> > I downloaded release and have this bug. Or I should compile the game from the source? Sure I will, but others will not, later will be a lot of question from people that just download the release.
> > I suggest just to update the release to 1.0.9.1 or so
> 
> Hi, @fheroes2bugs.
> 
> You can find it in the snapshot builds linked to on our installation guide.
> 
> https://github.com/ihhub/fheroes2/blob/master/docs/INSTALL.md#snapshots-latest-builds

Sure. But why no just release 1.0.9.1?

**Comment by ihhub** [comment]
@fheroes2bugs , we don't do releases just to fix a bug.

**Pull request merged**